Clevis 18 failed to build

### Describe the bug

The package `clevis` 18 on master failed to build.

### Steps To Reproduce
Steps to reproduce the behavior:
1. nix build github:nixos/nixpkgs#clevis

### Screenshots

![image](https://user-images.githubusercontent.com/11229748/190887437-b565b7a5-27d0-4589-ae8b-177098a51a72.png)

Full build log:
```
@nix { "action": "setPhase", "phase": "unpackPhase" }
unpacking sources
unpacking source archive /nix/store/nf0pl47abjiw6lrrvirrjd3cfcj4f86i-source
source root is source
@nix { "action": "setPhase", "phase": "patchPhase" }
patching sources
@nix { "action": "setPhase", "phase": "configurePhase" }
configuring
meson flags: --buildtype=plain         --libdir=/nix/store/dvbm2970xvrgd86clbzkdpy00vhj51ll-clevis-18/lib --libexecdir=/nix/store/dvbm2970xvrgd86clbzkdpy00vhj51ll-clevis-18/libexec         --bindir=/nix/store/dvbm2970xvrgd86clbzkdpy00vhj51ll-clevis-18/bin --sbindir=/nix/store/dvbm2970xvrgd86clbzkdpy00vhj51ll-clevis-18/sbin         --includedir=/nix/store/dvbm2970xvrgd86clbzkdpy00vhj51ll-clevis-18/include         --mandir=/nix/store/zfkc10dn2plp6jzhhi4mn893hgk50ccz-clevis-18-man/share/man --infodir=/nix/store/dvbm2970xvrgd86clbzkdpy00vhj51ll-clevis-18/share/info         --localedir=/nix/store/dvbm2970xvrgd86clbzkdpy00vhj51ll-clevis-18/share/locale         -Dauto_features=enabled         -Dwrap_mode=nodownload         --prefix=/nix/store/dvbm2970xvrgd86clbzkdpy00vhj51ll-clevis-18  
The Meson build system
Version: 0.63.1
Source dir: /build/source
Build dir: /build/source/build
Build type: native build
Project name: clevis
Project version: 18
C compiler for the host machine: gcc (gcc 11.3.0 "gcc (GCC) 11.3.0")
C linker for the host machine: gcc ld.bfd 2.38
Host machine cpu family: x86_64
Host machine cpu: x86_64
meson.build:39: WARNING: Consider using the built-in warning_level option instead of using "-Wall".
meson.build:39: WARNING: Consider using the built-in warning_level option instead of using "-Wextra".
meson.build:39: WARNING: Consider using the built-in werror option instead of using "-Werror".
Found pkg-config: /nix/store/q1bzihwpxznrc7kqg8n3ghx14b45sj0x-pkg-config-wrapper-0.29.2/bin/pkg-config (0.29.2)
Run-time dependency jansson found: YES 2.14
Run-time dependency jose found: YES 11
Program a2x found: YES (/nix/store/ixnmrx2hq1hrpa1dyj22rd5b64kp4ilx-asciidoc-full-10.2.0/bin/a2x)
Did not find CMake 'cmake'
Found CMake: NO
Run-time dependency bash-completion found: NO (tried pkgconfig and cmake)
src/bash/meson.build:7: WARNING: Will not install bash completion due to missing dependencies!
Run-time dependency luksmeta found: YES 9
Program pwmake found: YES (/nix/store/41bf2pz2habk7i5jmv1yjn2v9jh7q56c-libpwquality-1.4.4/bin/pwmake)
Run-time dependency libcryptsetup found: YES 2.5.0
Configuring clevis-luks-common-functions using configuration
Configuring clevis-luks-unbind using configuration
Run-time dependency systemd found: NO (tried pkgconfig and cmake)
Program /nix/store/dvbm2970xvrgd86clbzkdpy00vhj51ll-clevis-18/lib/systemd/systemd-reply-password /nix/store/dvbm2970xvrgd86clbzkdpy00vhj51ll-clevis-18/lib/systemd/systemd-reply-password /usr/lib/systemd/systemd-reply-password /usr/lib/systemd/systemd-reply-password found: NO
src/luks/systemd/meson.build:33: WARNING: Will not install systemd support due to missing dependencies!
Run-time dependency audit found: NO (tried pkgconfig and cmake)
Run-time dependency udisks2 found: NO (tried pkgconfig and cmake)
Run-time dependency gio-2.0 found: NO (tried pkgconfig and cmake)
src/luks/udisks2/meson.build:21: WARNING: Will not build udisks2 support due to missing dependencies!
Program jq found: NO
Program cryptsetup found: YES (/nix/store/lrl0xnbrqdnz8h3j4g93rhxkc1k4fm5k-cryptsetup-2.5.0-bin/bin/cryptsetup)
Configuring tests-common-functions using configuration
Program bind-wrong-pass-luks1 found: YES (/build/source/src/luks/tests/bind-wrong-pass-luks1)
Program bind-luks1 found: YES (/build/source/src/luks/tests/bind-luks1)
Program unbind-unbound-slot-luks1 found: YES (/build/source/src/luks/tests/unbind-unbound-slot-luks1)
Program unbind-luks1 found: YES (/build/source/src/luks/tests/unbind-luks1)
Program bind-key-file-non-interactive-luks1 found: YES (/build/source/src/luks/tests/bind-key-file-non-interactive-luks1)
Program bind-pass-with-newline-luks1 found: YES (/build/source/src/luks/tests/bind-pass-with-newline-luks1)
Program bind-pass-with-newline-keyfile-luks1 found: YES (/build/source/src/luks/tests/bind-pass-with-newline-keyfile-luks1)
Program bind-binary-keyfile-luks1 found: YES (/build/source/src/luks/tests/bind-binary-keyfile-luks1)
Program bind-already-used-luksmeta-slot found: YES (/build/source/src/luks/tests/bind-already-used-luksmeta-slot)
Program bad-sss found: YES (/build/source/src/luks/tests/bad-sss)
src/luks/tests/meson.build:49: WARNING: Will not run "clevis luks list" tests due to missing jq dependency
Program unlock-tang-luks1 found: YES (/build/source/src/luks/tests/unlock-tang-luks1)
Program assume-yes found: YES (/build/source/src/luks/tests/assume-yes)
Program regen-inplace-luks1 found: YES (/build/source/src/luks/tests/regen-inplace-luks1)
Program regen-not-inplace-luks1 found: YES (/build/source/src/luks/tests/regen-not-inplace-luks1)
Program report-tang-luks1 found: YES (/build/source/src/luks/tests/report-tang-luks1)
Program report-sss-luks1 found: YES (/build/source/src/luks/tests/report-sss-luks1)
Program edit-tang-luks1 found: YES (/build/source/src/luks/tests/edit-tang-luks1)
Program backup-restore-luks1 found: YES (/build/source/src/luks/tests/backup-restore-luks1)
Program pass-tang-luks1 found: YES (/build/source/src/luks/tests/pass-tang-luks1)
Program bind-wrong-pass-luks2 found: YES (/build/source/src/luks/tests/bind-wrong-pass-luks2)
Program bind-luks2 found: YES (/build/source/src/luks/tests/bind-luks2)
Program unbind-unbound-slot-luks2 found: YES (/build/source/src/luks/tests/unbind-unbound-slot-luks2)
Program unbind-luks2 found: YES (/build/source/src/luks/tests/unbind-luks2)
Program unlock-tang-luks2 found: YES (/build/source/src/luks/tests/unlock-tang-luks2)
Program assume-yes-luks2 found: YES (/build/source/src/luks/tests/assume-yes-luks2)
Program regen-inplace-luks2 found: YES (/build/source/src/luks/tests/regen-inplace-luks2)
Program regen-not-inplace-luks2 found: YES (/build/source/src/luks/tests/regen-not-inplace-luks2)
Program report-tang-luks2 found: YES (/build/source/src/luks/tests/report-tang-luks2)
Program report-sss-luks2 found: YES (/build/source/src/luks/tests/report-sss-luks2)
Program edit-tang-luks2 found: YES (/build/source/src/luks/tests/edit-tang-luks2)
Program backup-restore-luks2 found: YES (/build/source/src/luks/tests/backup-restore-luks2)
Program pass-tang-luks2 found: YES (/build/source/src/luks/tests/pass-tang-luks2)
Run-time dependency libcrypto found: YES 3.0.5
Program /build/source/src/pins/sss/pin-sss found: YES (/build/source/src/pins/sss/pin-sss)
Program /build/source/src/pins/sss/pin-test found: YES (/build/source/src/pins/sss/pin-test)
Program curl found: YES (/nix/store/vd4p2dirng26fkfk024l00hvz0z7f5ks-curl-7.84.0-bin/bin/curl)
Program /nix/store/dvbm2970xvrgd86clbzkdpy00vhj51ll-clevis-18/libexec/tangd-keygen /nix/store/dvbm2970xvrgd86clbzkdpy00vhj51ll-clevis-18/lib/tangd-keygen /nix/store/dvbm2970xvrgd86clbzkdpy00vhj51ll-clevis-18/libexec/tangd-keygen /usr/lib/tangd-keygen /usr/libexec/tangd-keygen found: NO
Program /nix/store/dvbm2970xvrgd86clbzkdpy00vhj51ll-clevis-18/libexec/tangd-update /nix/store/dvbm2970xvrgd86clbzkdpy00vhj51ll-clevis-18/lib/tangd-update /nix/store/dvbm2970xvrgd86clbzkdpy00vhj51ll-clevis-18/libexec/tangd-update /usr/lib/tangd-update /usr/libexec/tangd-update found: NO
Program /nix/store/dvbm2970xvrgd86clbzkdpy00vhj51ll-clevis-18/libexec/tangd /nix/store/dvbm2970xvrgd86clbzkdpy00vhj51ll-clevis-18/lib/tangd /nix/store/dvbm2970xvrgd86clbzkdpy00vhj51ll-clevis-18/libexec/tangd /usr/lib/tangd /usr/libexec/tangd found: NO
Program socat found: NO
Configuring tang-common-test-functions using configuration
Program pin-tang found: YES (/build/source/src/pins/tang/tests/pin-tang)
Program tang-validate-adv found: YES (/build/source/src/pins/tang/tests/tang-validate-adv)
Program default-thp-alg found: YES (/build/source/src/pins/tang/tests/default-thp-alg)
Program tpm2_createprimary found: YES (/nix/store/hyvrnxjzxbybmkfh8am0c52aagj7xfm1-tpm2-tools-5.2/bin/tpm2_createprimary)
Program tpm2_pcrread found: YES (/nix/store/hyvrnxjzxbybmkfh8am0c52aagj7xfm1-tpm2-tools-5.2/bin/tpm2_pcrread)
Program tpm2_createpolicy found: YES (/nix/store/hyvrnxjzxbybmkfh8am0c52aagj7xfm1-tpm2-tools-5.2/bin/tpm2_createpolicy)
Program tpm2_create found: YES (/nix/store/hyvrnxjzxbybmkfh8am0c52aagj7xfm1-tpm2-tools-5.2/bin/tpm2_create)
Program tpm2_load found: YES (/nix/store/hyvrnxjzxbybmkfh8am0c52aagj7xfm1-tpm2-tools-5.2/bin/tpm2_load)
Program tpm2_unseal found: YES (/nix/store/hyvrnxjzxbybmkfh8am0c52aagj7xfm1-tpm2-tools-5.2/bin/tpm2_unseal)
Program pin-tpm2 found: YES (/build/source/src/pins/tpm2/pin-tpm2)
Program update-initramfs found: NO
src/initramfs-tools/meson.build:15: WARNING: Will not install initramfs-tools module due to missing dependencies!
Build targets in project: 16

clevis 18

  User defined options
    auto_features: enabled
    bindir       : /nix/store/dvbm2970xvrgd86clbzkdpy00vhj51ll-clevis-18/bin
    buildtype    : plain
    includedir   : /nix/store/dvbm2970xvrgd86clbzkdpy00vhj51ll-clevis-18/include
    infodir      : /nix/store/dvbm2970xvrgd86clbzkdpy00vhj51ll-clevis-18/share/info
    libdir       : /nix/store/dvbm2970xvrgd86clbzkdpy00vhj51ll-clevis-18/lib
    libexecdir   : /nix/store/dvbm2970xvrgd86clbzkdpy00vhj51ll-clevis-18/libexec
    localedir    : /nix/store/dvbm2970xvrgd86clbzkdpy00vhj51ll-clevis-18/share/locale
    mandir       : /nix/store/zfkc10dn2plp6jzhhi4mn893hgk50ccz-clevis-18-man/share/man
    prefix       : /nix/store/dvbm2970xvrgd86clbzkdpy00vhj51ll-clevis-18
    sbindir      : /nix/store/dvbm2970xvrgd86clbzkdpy00vhj51ll-clevis-18/sbin
    wrap_mode    : nodownload

Found ninja-1.11.0 at /nix/store/s22ai6fkaa5sq35wy74znc0hxfqanhvb-ninja-1.11.0/bin/ninja
meson: enabled parallel building
@nix { "action": "setPhase", "phase": "buildPhase" }
building
build flags: -j8 -l8
[1/20] Compiling C object src/pins/sss/clevis-decrypt-sss.p/sss.c.o
FAILED: src/pins/sss/clevis-decrypt-sss.p/sss.c.o 
gcc -Isrc/pins/sss/clevis-decrypt-sss.p -Isrc/pins/sss -I../src/pins/sss -I/nix/store/x9l2dhc2hhdj96sczxxz7nnksfk08wlf-jansson-2.14/include -I/nix/store/gd8xbg6y246ri68pz4qkwdczydwm6liq-openssl-3.0.5-dev/include -I/nix/store/j49rml3k3c2xq72y7gz0v9c2k30iz5xd-jose-11-dev/include -fdiagnostics-color=always -D_FILE_OFFSET_BITS=64 -Wall -Winvalid-pch -std=c99 -Wall -Wextra -Werror -Wstrict-aliasing -Wchar-subscripts -Wformat-security -Wmissing-declarations -Wmissing-prototypes -Wnested-externs -Wpointer-arith -Wshadow -Wsign-compare -Wstrict-prototypes -Wtype-limits -Wunused-function -Wno-missing-field-initializers -Wno-unused-parameter -Wno-unknown-pragmas -D_DEFAULT_SOURCE -D_POSIX_C_SOURCE=200112L '-DBINDIR="/nix/store/dvbm2970xvrgd86clbzkdpy00vhj51ll-clevis-18/bin"' '-DCLEVIS_USER="clevis"' '-DCLEVIS_GROUP="clevis"' -MD -MQ src/pins/sss/clevis-decrypt-sss.p/sss.c.o -MF src/pins/sss/clevis-decrypt-sss.p/sss.c.o.d -o src/pins/sss/clevis-decrypt-sss.p/sss.c.o -c ../src/pins/sss/sss.c
In file included from ../src/pins/sss/sss.c:41:
../src/pins/sss/sss.c: In function 'sss_point':
../src/pins/sss/sss.c:217:9: error: void value not ignored as it ought to be
  217 |     if (BN_zero(yy) <= 0)
      |         ^~~~~~~
../src/pins/sss/sss.c: In function 'sss_recover':
../src/pins/sss/sss.c:275:9: error: void value not ignored as it ought to be
  275 |     if (BN_zero(k) <= 0)
      |         ^~~~~~~
../src/pins/sss/sss.c:306:17: error: void value not ignored as it ought to be
  306 |             if (BN_zero(tmp) <= 0)
      |                 ^~~~~~~
[2/20] Compiling C object src/pins/sss/clevis-encrypt-sss.p/sss.c.o
FAILED: src/pins/sss/clevis-encrypt-sss.p/sss.c.o 
gcc -Isrc/pins/sss/clevis-encrypt-sss.p -Isrc/pins/sss -I../src/pins/sss -I/nix/store/x9l2dhc2hhdj96sczxxz7nnksfk08wlf-jansson-2.14/include -I/nix/store/gd8xbg6y246ri68pz4qkwdczydwm6liq-openssl-3.0.5-dev/include -I/nix/store/j49rml3k3c2xq72y7gz0v9c2k30iz5xd-jose-11-dev/include -fdiagnostics-color=always -D_FILE_OFFSET_BITS=64 -Wall -Winvalid-pch -std=c99 -Wall -Wextra -Werror -Wstrict-aliasing -Wchar-subscripts -Wformat-security -Wmissing-declarations -Wmissing-prototypes -Wnested-externs -Wpointer-arith -Wshadow -Wsign-compare -Wstrict-prototypes -Wtype-limits -Wunused-function -Wno-missing-field-initializers -Wno-unused-parameter -Wno-unknown-pragmas -D_DEFAULT_SOURCE -D_POSIX_C_SOURCE=200112L '-DBINDIR="/nix/store/dvbm2970xvrgd86clbzkdpy00vhj51ll-clevis-18/bin"' '-DCLEVIS_USER="clevis"' '-DCLEVIS_GROUP="clevis"' -MD -MQ src/pins/sss/clevis-encrypt-sss.p/sss.c.o -MF src/pins/sss/clevis-encrypt-sss.p/sss.c.o.d -o src/pins/sss/clevis-encrypt-sss.p/sss.c.o -c ../src/pins/sss/sss.c
In file included from ../src/pins/sss/sss.c:41:
../src/pins/sss/sss.c: In function 'sss_point':
../src/pins/sss/sss.c:217:9: error: void value not ignored as it ought to be
  217 |     if (BN_zero(yy) <= 0)
      |         ^~~~~~~
../src/pins/sss/sss.c: In function 'sss_recover':
../src/pins/sss/sss.c:275:9: error: void value not ignored as it ought to be
  275 |     if (BN_zero(k) <= 0)
      |         ^~~~~~~
../src/pins/sss/sss.c:306:17: error: void value not ignored as it ought to be
  306 |             if (BN_zero(tmp) <= 0)
      |                 ^~~~~~~
[3/20] Compiling C object src/pins/sss/clevis-decrypt-sss.p/clevis-decrypt-sss.c.o
[4/20] Compiling C object src/pins/sss/clevis-encrypt-sss.p/clevis-encrypt-sss.c.o
[5/20] Generating clevis-luks-unbind.1 with a custom command
[6/20] Generating clevis-luks-list.1 with a custom command
[7/20] Generating clevis-luks-unlockers.7 with a custom command
[8/20] Generating clevis-luks-bind.1 with a custom command
ninja: build stopped: subcommand failed.
```

### Notify maintainers

@fpletz 

### Metadata

```console
 - system: `"x86_64-linux"`
 - host os: `Linux 5.15.67, NixOS, 22.11 (Raccoon), 22.11.20220912.5f326e2`
 - multi-user?: `yes`
 - sandbox: `yes`
 - version: `nix-env (Nix) 2.11.0`
 - channels(root): `"nixos-21.11pre318500.51bcdc4cdaa, home-manager"`
 - channels(yinfeng): `""`
 - nixpkgs: `/nix/store/8s4q1xjnpkklhw3lgcxlvab9g6kln4qk-source`
```

--- a/pkgs/tools/security/clevis/default.nix
+++ b/pkgs/tools/security/clevis/default.nix
@@ -1,6 +1,7 @@
 { lib
 , stdenv
 , fetchFromGitHub
+, fetchurl
 , meson
 , ninja
 , pkg-config
@@ -28,6 +29,14 @@ stdenv.mkDerivation rec {
     rev = "v${version}";
     sha256 = "sha256-m1UhyjD5ydSgCTBu6sECLlxFx0rnQxFnBA7frbdUqU8=";
   };
+
+  patches = [
+    # sss: use BN_set_word(x, 0) instead of BN_zero(), fixes build issue with different versions of openssl
+    (fetchurl {
+      url = "https://github.com/latchset/clevis/commit/ee1dfedb9baca107e66a0fec76693c9d479dcfd9.patch";
+      sha256 = "sha256-GeklrWWlAMALDLdnn6+0Bi0l+bXrIbYkgIyI94WEybM=";
+    })
+  ];
 
   postPatch = ''
     for f in $(find src/ -type f); do

